gcloud-java-compute/src/test/java/com/google/gcloud/compute/ComputeImplTest.java

```diff
@@ -695,20 +832,355 @@ public void testGetLicenseFromId() {
+    String cursor = "cursor";
+    compute = options.service();
+    ImmutableList<Operation> operationList = ImmutableList.of(globalOperation, globalOperation);
+    Tuple<String, Iterable<com.google.api.services.compute.model.Operation>> result =
+        Tuple.of(cursor, Iterables.transform(operationList,
+            new Function<Operation, com.google.api.services.compute.model.Operation>() {
+              @Override
+              public com.google.api.services.compute.model.Operation apply(Operation operation) {
+                return operation.toPb();
+              }
+            }));
+    EasyMock.expect(computeRpcMock.listGlobalOperations(OPERATION_LIST_OPTIONS)).andReturn(result);
+    EasyMock.replay(computeRpcMock);
+    Page<Operation> page = compute.listGlobalOperations(OPERATION_LIST_MAX_RESULTS,
+        OPERATION_LIST_PAGE_TOKEN, OPERATION_LIST_FILTER);
+    assertEquals(cursor, page.nextPageCursor());
+    assertArrayEquals(operationList.toArray(), Iterables.toArray(page.values(), Operation.class));
+  }
+
+  @Test
+  public void testDeleteGlobalOperation_True() {
+    EasyMock.expect(computeRpcMock.deleteGlobalOperation(GLOBAL_OPERATION_ID.operation()))
+        .andReturn(true);
+    EasyMock.replay(computeRpcMock);
+    compute = options.service();
+    assertTrue(compute.delete(GLOBAL_OPERATION_ID));
+  }
+
+  @Test
+  public void testDeleteGlobalOperation_False() {
+    EasyMock.expect(computeRpcMock.deleteGlobalOperation(GLOBAL_OPERATION_ID.operation()))
+        .andReturn(false);
+    EasyMock.replay(computeRpcMock);
+    compute = options.service();
+    assertFalse(compute.delete(GLOBAL_OPERATION_ID));
+  }
+
+  @Test
+  public void testGetRegionOperation() {
+    EasyMock.expect(computeRpcMock.getRegionOperation(REGION_OPERATION_ID.region(),
+            REGION_OPERATION_ID.operation(), EMPTY_RPC_OPTIONS))
+        .andReturn(regionOperation.toPb());
+    EasyMock.replay(computeRpcMock);
+    compute = options.service();
+    Operation operation = compute.get(REGION_OPERATION_ID);
+    assertEquals(regionOperation, operation);
+  }
+
+  @Test
+  public void testGetRegionOperationWithSelectedFields() {
+    Capture<Map<ComputeRpc.Option, Object>> capturedOptions = Capture.newInstance();
+    EasyMock.expect(computeRpcMock.getRegionOperation(eq(REGION_OPERATION_ID.region()),
+            eq(REGION_OPERATION_ID.operation()), capture(capturedOptions)))
+        .andReturn(regionOperation.toPb());
+    EasyMock.replay(computeRpcMock);
+    compute = options.service();
+    Operation operation = compute.get(REGION_OPERATION_ID, OPERATION_OPTION_FIELDS);
+    String selector = (String) capturedOptions.getValue().get(OPERATION_OPTION_FIELDS.rpcOption());
+    assertTrue(selector.contains("selfLink"));
+    assertTrue(selector.contains("id"));
+    assertTrue(selector.contains("description"));
+    assertEquals(23, selector.length());
+    assertEquals(regionOperation, operation);
```

Comment:
The purpose of this is to assert that the object passed by the spi layer is correctly wrapped and passed on. It might be weird but this does the job. Creating a partial operation object adds nothing to the test and given that the file is already 1.2k lines long (and we tested only 1/10 of Compute resources) I would tend to avoid creating other objects if not strictly needed.